main.cpp

```diff
@@ -54,48 +54,73 @@ Purpose:  This project will show you the difference between member functions and
 
 #include <iostream>
 #include <string>
+
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    T(int v, const char* n) : value(v), name(n) {}  
+    int value;                                     
+    std::string name;                              
 };
 
-struct <#structName1#>                                //4
+struct Comparison                                
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare( T* a, T* b ) 
     {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
+        if( a && b )
```

Comment:
Be explicit when checking pointes and whether or not they `== nullptr`
while this is concise, the intent regarding what you're checking is not as obvious as it could be.
- [ ] revise